app/assets/javascripts/components/overview/overview_handler.jsx

```diff
@@ -47,13 +50,29 @@ const Overview = createReactClass({
 
   componentDidMount() {
     ServerActions.fetch('timeline', this.props.course_id);
+    this.timeout = this.poll();
     return ServerActions.fetch('tags', this.props.course_id);
   },
 
+  componentWillUnmount() {
+    if (this.timeout) {
+      clearInterval(this.timeout);
+    }
+  },
+
   storeDidChange() {
     return this.setState(getState());
   },
 
+  poll() {
+    if (this.state.course.level === 'Editathon') {
+      return setInterval(() => CourseActions.updateCourse(), POLL_INTERVAL);
+    }
+    return null;
+  },
+
+  timeout: null,
```

Comment:
Doing it like this seems weird. Why keep the timeout on the class directly, instead of in state?

Reply:
IMO, it need not be in state because, it is not going to change once its assigned. Also, I found `this.timeout` more semantic because `Overview` has a timeout, *not the state* of `Overview`.